python/pyspark/ml/clustering.py

```diff
@@ -201,7 +203,74 @@ def computeCost(self, dataset):
         """
         return self._call_java("computeCost", dataset)
 
+    @since("2.0.0")
+    def hasSummary(self):
+        """
+        Return true if there exists summary of model.
+        """
+        return self.summary is not None
+
+    @property
+    @since("2.0.0")
+    def summary(self):
+        """
+        Gets summary of model on training set.
```

Comment:
Maybe good to add a notice this causes an exception if there is no summary present (as done in the scala docs).